docs/design/proposals/status-conditions.md

```diff
@@ -198,18 +198,84 @@ Things to note:
 
 ##### Resource Graph Instance
 
-This one is currently pretty tricky because it accepts conditions to be defined
-inside the RGD Instance. But at minimum and by default the controller adds:
+The Resource Graph Instance currently uses a single condition:
 
 - InstanceSynced
 
-To be able to use a condition manager, we would need a signal from the provided
-status shape to understand if the defined condition is a dependent condition or
-just informational.
+Based on analysis of the instance reconciler, the major sections of work can be
+broken down into the following sub-conditions that align with the reconciliation
+phases:
+
+```
+Ready
+├─ InstanceManaged - Instance finalizers and labels are properly set
+├─ GraphResolved - Runtime graph has been created and resources resolved
+└─ ResourcesReady - All resources in the graph are created and ready
+```
+
+The reconciler performs the following major work sections:
+
+1. **Instance Setup**: Setting managed state and finalizers (InstanceManaged)
+2. **Runtime Creation**: Creating the ResourceGraphDefinition runtime and resolving resources (GraphResolved)
+3. **Resource Reconciliation**: Processing each resource in topological order - creation, updates, and readiness checks (ResourcesReady)
+4. **Instance Deletion**: Deleting resources in reverse topological order
 
 A similar example applies to Resource Graph Instance as shown under Resource
 Graph Definition.
 
+`Current Instance Status`
+
+```
+status:
+  state: ACTIVE
+  conditions:
+  - lastTransitionTime: "2025-08-07T21:31:02Z"
+    message: Instance reconciled successfully
+    reason: ReconciliationSucceeded
+    status: "True"
+    type: InstanceSynced
+```
+
+`Proposed Update`
+
+```
+status:
+  state: ACTIVE
+  conditions:
+  - lastTransitionTime: "2025-08-07T21:31:02Z"
+    message: ""
+    observedGeneration: 1
+    reason: "AllReady"
+    status: "True"
+    type: Ready
+  - lastTransitionTime: "2025-08-07T21:31:02Z"
+    message: Instance is properly managed with finalizers and labels
+    observedGeneration: 1
+    reason: "Managed"
+    status: "True"
+    type: InstanceManaged
+  - lastTransitionTime: "2025-08-07T21:31:02Z"
+    message: Runtime graph created and all resources resolved
+    observedGeneration: 1
+    reason: "Resolved"
+    status: "True"
+    type: GraphResolved
+  - lastTransitionTime: "2025-08-07T21:31:02Z"
+    message: All resources are created and ready
+    observedGeneration: 1
+    reason: "AllResourcesReady"
+    status: "True"
+    type: ResourcesReady
+```
+
+Things to note:
+
+- The messages have been customized to reflect the specific phase of reconciliation
+- ObservedGeneration is added and used in all conditions
+- The condition Types reflect the major work sections in the instance reconciler:
+  - InstanceSynced → InstanceManaged + GraphResolved + ResourcesReady structure
```

Comment:
wondering where InstanceSynced is in your overview now? do you overwrite it with InstanceManaged?

In general I must admit that to me InstanceSynced is much more intuititive but I guess thats subjective. Thats because for me the resource being managed is not directly a correlation to it being correctly synced with the instance graph

Reply:
being synced is just one small part of the sub-condition of `ResourcesReady`. ResourcedReady would be they are synced and ready, they can't be ready without being synced + the observed generation match generation.